doc/domains.rst

```diff
@@ -1103,6 +1103,22 @@ The JavaScript Domain
 
 The JavaScript domain (name **js**) provides the following directives:
 
+.. rst:directive:: .. js:module:: name
+
+   This directive sets the module name for object declarations that follow
+   after. The module name is used in the global module index and in cross
+   references. This directive does not create an object heading like
+   :rst:dir:`py:class` would, for example.
+
+   By default, this directive will create a linkable entity and will cause an
+   entry in the global module index, unless the ``noindex`` option is specified.
+   If this option is specified, the directive will only update the current
+   module name.
+
+   To clear the current module, set the module name to ``null`` or ``None``
```

Comment:
This is new concept. So we need to discuss carefully; about parameter name, supported domains and so on.
Could you seperate this to another PR? We'll review it after merged this.